mlfinlab/feature_importance/fingerpint.py

```diff
@@ -0,0 +1,259 @@
+"""
+Implementation of an algorithm described in Yimou Li, David Turkington, Alireza Yazdani
+'Beyond the Black Box: An Intuitive Approach to Investment Prediction with Machine Learning'
+(https://jfds.pm-research.com/content/early/2019/12/11/jfds.2019.1.023)
+"""
+
+from abc import ABC, abstractmethod
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.linear_model import LinearRegression
+
+
+# pylint: disable=invalid-name
+
+class AbstractModelFingerprint(ABC):
+    """
+    Model fingerprint constructor.
+    This is an abstract class, use RegressionModelFingerprint or ClassificationModelFingerprint.
+
+    :param model: (object) trained regression model
+    :param X: (pd.DataFrame) of features
+    :param num_values: (int) number of values used to estimate feature effect
+    """
+
+    def __init__(self, model: object, X: pd.DataFrame, num_values=50):
+        """
+        Model fingerprint constructor.
+        """
+
+        self.X = X
+        self.model = model
+
+        # Effects containers
+        self.linear_effect = None
+        self.non_linear_effect = None
+        self.pair_wise_effect = None
+
+        self.num_values = num_values
+
+        # Get possible feature values (Step 1)
+        self.feature_values = {}  # Dictionary of feature values range used to build dependence functions
+        for feature in self.X.columns:
+            values = []
+            for q in np.linspace(0, 1, self.num_values):
+                values.append(np.quantile(self.X[feature], q=q))
+            self.feature_values[feature] = np.array(values)
+
+        # DataFrame with partial dependence function values for columns from X (f_k)
+        self.ind_partial_dep_functions = pd.DataFrame(index=list(range(self.num_values)), columns=self.X.columns)
+
+        self.feature_column_position_mapping = dict(zip(self.X.columns, range(0, self.X.shape[1])))
+
+        # Get partial dependency functions
+        self._get_individual_partial_dependence()
+
+    @abstractmethod
+    def _get_model_predictions(self, X_):
+        """
+        Get model predictions based on problem type (predict for regression, predict_proba for classification)
+
+        :param X_: (np.array) feature set
+        :return: (np.array) of predictions
+        """
+        raise NotImplementedError('Must implement _get_model_predictions')
+
+    def _get_individual_partial_dependence(self) -> None:
+        """
+        Get individual partial dependence function values for each column.
+
+        :return: None
```

Comment:
Remove return None from methods that don't return anything. That way the documentation is less verbose and a little more clean.

Reply:
None type here